airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py

```diff
@@ -32,7 +32,7 @@
 
 class SendgridStream(HttpStream, ABC):
     url_base = "https://api.sendgrid.com/v3/"
-    primary_key = "id"
+    primary_key = "email"
```

Comment:
I don't think this is correct, some streams don't have `email` attribute, but have `id`, I think you need to check each stream and override primary_key = "email" where necessary, WDYT?

Reply:
all incremental streams at the moment use `email` other only have `full_refresh` enabled.

Reply:
in such case maybe it is better to leave primary_key=id, and override Incremental base class?